### PR TITLE
Task/frontend248 fix history browse page skeleton

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
[Skeletons appears on the empty History page after refreshing it](https://trello.com/c/rZ8RThYf/248-skeletons-appears-on-the-empty-history-page-after-refreshing-it)

File: backend/core/src/shared/helpers/pagination/use-pagination.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export type PaginationHelperInputType = {
+  allDataLength: number;
+  pageNumber: number | typeof NaN;
```
Comment (droztoff) [inline backend/core/src/shared/helpers/pagination/use-pagination.ts]: This solution looks weird, explain why you did it? (number | number)

File: frontend/src/pages/browse-page/browse-page.tsx [hunk 2]
```diff
@@ -69,14 +75,14 @@ const BrowsePage: FC = () => {
...
-        {videoData.dataStatus === DataStatus.PENDING ? generateHistorySkeletons(isLightTheme) : null}
+        {videoData.dataStatus === DataStatus.PENDING && generateBrowsePageSkeleton(isLightTheme, lastListLength)}
         {popularVideos.list.map((video) => {
           return <VideoCardMain key={video.id} video={video} isLightTheme={isLightTheme} />;
         })}
         <div ref={sentryRef}>
-          {videoData.dataStatus === DataStatus.PENDING && popularVideos.list.length > 0 ? (
-            <Loader hCentered={true} vCentered={true} spinnerSize={'md'} />
-          ) : null}
+          {videoData.dataStatus === DataStatus.PENDING && popularVideos.list.length > 0
```
Comment (droztoff) [inline frontend/src/pages/browse-page/browse-page.tsx]: You can wrap the expression in parentheses and you won't need to use the ternary operator

File: frontend/src/pages/history-page/history-page.tsx [hunk 3]
```diff
@@ -24,28 +24,29 @@ const HistoryPage: FC = () => {
     return state.theme.isLightTheme;
   });
 
+  const historyData = useAppSelector((state) => {
+    return state.history.data;
+  });
+
+  const { isFirstHistoryPageLoad, currentPage, lastPage, dataStatus } = historyData;
```
Comment (droztoff) [inline frontend/src/pages/history-page/history-page.tsx]: you can apply destructuring on the line where you get historyData

File: frontend/src/pages/browse-page/browse-page.tsx (hunk 2 above)
Comment (ghost) [reply frontend/src/pages/browse-page/browse-page.tsx]: done

File: frontend/src/pages/history-page/history-page.tsx (hunk 3 above)
Comment (ghost) [reply frontend/src/pages/history-page/history-page.tsx]: done

File: backend/core/src/shared/helpers/pagination/use-pagination.ts (hunk 1 above)
Comment (ghost) [reply backend/core/src/shared/helpers/pagination/use-pagination.ts]: remove typeof NaN

**Review by droztoff — APPROVED** [review]

**Pull request merged**